expbackoff/backoff.go

```diff
@@ -0,0 +1,100 @@
+package expbackoff
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	prom "github.com/prometheus/client_golang/prometheus"
+)
+
+// BackOff is wrapper that provides retry mechanism.
+type BackOff struct {
+	counter prom.Counter
+	backoff backoff.BackOff
+	timeout time.Duration
+}
+
+type RetriesCfg struct {
+	Limited bool
+	Limit   uint64
+}
+
+// New instance of BackoffTimeout.
+func New(
+	ctx context.Context,
+	counter prom.Counter,
+	timeout time.Duration,
+	retriesCfg RetriesCfg,
+	opts ...Option,
+) *BackOff {
+	var expBackoff backoff.BackOff
+	expBackoff = backoff.WithContext(new(opts...), ctx)
+	if retriesCfg.Limited {
+		expBackoff = backoff.WithMaxRetries(expBackoff, retriesCfg.Limit)
+	}
+
+	return &BackOff{
+		counter: counter,
+		backoff: expBackoff,
+		timeout: timeout,
+	}
+}
+
+// Exec processes gives lambda.
+func (b *BackOff) Exec(ctx context.Context, executor func(ctx context.Context) error) error {
```

Comment:
If BackOff is a wrapper, then use function names as it wrapped object

Reply:
`RetryWithMetrics` seems appropriate.